src/components/doc-viewer/doc-viewer.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  Component,
+  NgModule,
+  ModuleWithProviders,
+  Input,
+  ElementRef,
+  ComponentFactoryResolver,
+  ViewContainerRef
+} from '@angular/core';
+import {Http} from '@angular/http';
+import {DomPortalHost, ComponentPortal} from '@angular/material';
+import {MaterialIoExampleViewer} from '../example-viewer/example-viewer';
+
+
+@Component({
+  selector: 'material-io-doc-viewer',
+  template: 'Loading document...',
+})
+export class MaterialIoDocViewer {
+  constructor(private _componentFactoryResolver: ComponentFactoryResolver,
+              private _elementRef: ElementRef,
+              private _http: Http,
+              private _viewContainerRef: ViewContainerRef) {}
+
+  /** The url of the document to display. */
+  @Input('document-url')
```

Comment:
I've seen this in the material2 repo too, should we switch those to also be camelCase?